### ISSUE TITLE
False warnings with cargo doc.

### ISSUE THREAD

**Issue by prataprc** [rust-lang/rust#58745]
First raised this issue with ``rust-random/rand`` https://github.com/rust-random/rand/issues/737. But the documentation links seem to work fine. So may be it is false-warning raised by ``cargo doc`` ?

How to reproduce ?

```bash
$ git clone git@github.com:bnclabs/llrb-index.git
$ cd llrb-index

$ rustup which cargo
<path>/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo

$ cargo doc
```

**Comment by GuillaumeGomez** [comment]
I'll take a look.

**Comment by QuietMisdreavus** [comment]
I took a closer look at this yesterday and today, and found some of the warnings that were in fact spurious - the ones that corresponded to a link that was actually resolved on the item's page. I've opened https://github.com/rust-lang/rust/pull/58972 to fix those.

However, there were some left that were not spurious in the same way. These all tripped a warning because the links can resolve in the original location in the original crate, but not the re-exported location in the facade crate. (The links are resolved all over again each time the item re-exports into a full page - the information isn't saved across crates, so as a hack we use the location of the re-export to resolve links.) Once https://github.com/rust-lang/rust/pull/58972 is merged, the remaining warnings will need more effort to fix.

**Comment by dhardy** [comment]
Thanks for looking into this.

The latter issues sound tricky to solve — correct documentation for an item presented differently in two different places. I wonder if `doc` attributes could be made optional depending on which crate the item is being exported from? Ideally here we'd just want a different path for `test_timer` but significantly different docs for `RngCore`.

**Comment by GuillaumeGomez** [comment]
@QuietMisdreavus and I handled the issue differently: I don't show errors on items that are outside of the current crate and they prevent to read non-local items (so no errors since it's not parsed).

**Comment by jyn514** [comment]
> These all tripped a warning because the links can resolve in the original location in the original crate, but not the re-exported location in the facade crate.

This sounds like it will be fixed by #73101, but I haven't tested on these exact crates.

**Comment by jyn514** [comment]
I documented rand without warnings on the latest nightly, and the upstream issue has been closed (https://github.com/rust-random/rand/issues/737), so I'm going to close this as well. Feel free to re-open if you still have issues!
